### PR TITLE
fix: check api key access and query the asset inventory accordingly

### PR DESCRIPTION
**Pull request by Biglr**
fix: formatting
fix: update modules

### CODE REVIEW

File: src/DataSource.ts [hunk 1]
```diff
@@ -97,31 +95,65 @@ export class DataSource extends DataSourceApi<AkenzaQuery, AkenzaDataSourceConfi
...
-                        },
-                        error(error: FetchError) {
-                            reject(DataSource.generateErrorMessage(error))
-                        }
-                    });
-            });
+        const organization = await this.getOrganization();
+        const access = await this.checkWorkspaceAccess(organization.id);
+
+        const params: AccessCheckParams = {
```
Comment (mariusgiger) [inline src/DataSource.ts:105]: the type of this should likely be AssetFilterParams

File: src/DataSource.ts [hunk 2]
```diff
@@ -97,31 +95,65 @@ export class DataSource extends DataSourceApi<AkenzaQuery, AkenzaDataSourceConfi
...
+                        resolve(response.data.content)
+                    },
+                    error(error: FetchError) {
+                        reject(DataSource.generateErrorMessage(error))
+                    }
+                });
+        });
+    }
+
+    async checkWorkspaceAccess(organizationId: string): Promise<AccessCheckResponse> {
```
Comment (mariusgiger) [inline src/DataSource.ts:137]: did you find a way to cache this?

**Review by mariusgiger — COMMENTED** [review]
looks good, a small comment

File: src/DataSource.ts (hunk 1 above)
Comment (Biglr) [reply src/DataSource.ts:105]: ah yes, the old name confusion ^^

File: src/DataSource.ts (hunk 2 above)
Comment (Biglr) [reply src/DataSource.ts:137]: hm, completely forgot about that aspect...

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**